zipline/finance/controls.py

```diff
@@ -99,6 +99,38 @@ def validate(self,
         self.orders_placed += 1
 
 
+class RestrictedListOrder(TradingControl):
+    """
+    TradingControl representing a restricted list of securities that
+    cannot be ordered by the algorithm.
+    """
+
+    def __init__(self, restricted_list):
+        """
+        restricted list can be an iterable, or
+        a callable that returns an iterable for dynamic restrictions.
+        """
+
+        super(RestrictedListOrder, self).__init__()
+        # if passed a callable, call it, otherwise
+        # wrap what we were passed in a lambda
+        self.restricted_list = \
```

Comment:
What benefit is gained by allowing the user to pass a callable here?  Since the function has to take no arguments, the only way this can ever return something different on different calls is if it's closing over global mutable state, which seems like an idiom we want to discourage (in particular, this is the doomsday scenario for serialization).